consensus/tendermint/core/upon_condition_test.go

```diff
@@ -570,6 +570,127 @@ func TestOldProposal(t *testing.T) {
 		assert.Equal(t, clientLockedValue, c.lockedValue)
 		assert.Equal(t, clientLockedValue, c.validValue)
 	})
+
+	// line 28 check upon condition on prevote handler.
+	/*
+		  Please refer to the discussion history: https://github.com/clearmatics/autonity/pull/615
+		This test case address the scenario laid out below.
```

Comment:
This is a bit misleading, since the scenario is now handled.

```suggestion
	Please refer to the discussion history: https://github.com/clearmatics/autonity/pull/615
	This test case validates the need for the upon condition defined in line 28 of the tendermint
	whitepaper, which addresses the scenario laid out below. 
```